wagtail/contrib/redirects/tests/test_redirects_api.py

```diff
@@ -0,0 +1,84 @@
+from django.test import TestCase
+from django.urls import reverse
+
+from wagtail.contrib.redirects.models import Redirect
+from wagtail.models import Page, Site
+
+
+class TestRedirectsAPI(TestCase):
+    def setUp(self):
+        self.example_home = Page.objects.get(slug="home").add_sibling(
+            instance=Page(title="Example Homepage", slug="example-home")
+        )
+        self.example_page = self.example_home.add_child(
+            instance=Page(title="Example Page", slug="example-page")
+        )
+        self.example_site = Site.objects.create(
+            hostname="example", root_page=self.example_home
+        )
+
+        Redirect.objects.create(
+            old_path="/hello-world",
+            site=self.example_site,
+            redirect_link="https://www.example.com/hello-world/",
+        )
+
+        Redirect.objects.create(
+            old_path="/good-work",
+            site=self.example_site,
+            redirect_link="https://www.example.com/hello-world/",
+        )
+
+        Redirect.add_redirect(
+            old_path="/hello-world", redirect_to="https://www.example.net/new-world/"
+        )
+
+        Redirect.add_redirect(
+            old_path="/old-example", redirect_to=self.example_home, is_permanent=False
+        )
+
+        Redirect.add_redirect(
+            old_path="/old-example?bar=foo&foo=bar",
+            redirect_to=self.example_page,
+            is_permanent=False,
+        )

```

Comment:
We should add a test for just the index listing also, using `find` is not the only thing added with this new API endpoint